sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Span, SpanOptions, SpanKind } from "@opentelemetry/api";
+import { getTracer } from "../src/tracerProxy";
+import { OperationTracingOptions } from "./interfaces";
+
+/**
+ * Arguments for `createSpanFunction` that allow you to specify the
+ * prefix for each created span as well as the `az.namespace` attribute.
+ *
+ * @hidden
+ */
+export interface CreateSpanFunctionArgs {
+  /**
+   * Package name prefix
+   */
+  packagePrefix: string;
```

Comment:
Just a suggestion for a future PR..
It could be amazing if this can enforce how the prefix should be rather than just a string.
Something similar to the TS types parsing magic that @joheredi had presented for core-client.

Same is applicable for the "namespace" attribute.

Reply:
The trick here is that the domain for these are not known - they're per package. It would be interesting if the slugs that are here are standardized - I'm still a bit unaware how they're "allocated". It's pretty clear, looking at most of them, that they're being originated in .net (or at least, they follow their naming convention _and_ part of their namespaces).

However, we don't have to do too much here - you'll typically call this exactly once in your code and store off the created function and call _that_ multiple times. So it's not going to lead to duplication.